packages/varlet-ui/src/rate/__tests__/index.spec.js

```diff
@@ -72,64 +43,83 @@ test('test rate validation', async () => {
-  expect(wrapper.html()).toMatchSnapshot()
-
   await wrapper.find('.var-rate__content').trigger('click')
   await delay(16)
   expect(wrapper.find('.var-form-details__message').exists()).toBeFalsy()
-  expect(wrapper.html()).toMatchSnapshot()
-
   wrapper.vm.reset()
   await delay(16)
   expect(wrapper.props('modelValue')).toBe(0)
 
   wrapper.unmount()
 })
 
-test('test rate disabled & readonly', async () => {
-  const onChange = jest.fn()
-
-  const wrapper = mount(VarRate, {
-    props: {
-      disabled: true,
-      readonly: false,
-      onChange,
-    },
+describe('test rate component props', () => {
+  test('test rate half', async () => {
+    const onChange = jest.fn()
+    const onUpdateModelValue = jest.fn((value) => wrapper.setProps({ modelValue: value }))
+
+    const wrapper = mount(VarRate, {
+      props: {
+        half: true,
+        modelValue: 0,
+        onChange,
+        'onUpdate:modelValue': onUpdateModelValue,
+      },
+    })
+
+    await trigger(wrapper.find('.var-rate__content'), 'click')
+    expect(onChange).toHaveBeenCalledTimes(1)
+    expect(onUpdateModelValue).toHaveBeenCalledTimes(1)
+    expect(wrapper.props('modelValue')).toBe(0.5)
+
+    wrapper.unmount()
   })
 
-  await wrapper.find('.var-rate__content').trigger('click')
-  expect(wrapper.html()).toMatchSnapshot()
-  expect(onChange).toHaveBeenCalledTimes(0)
+  test('test rate disabled & readonly', async () => {
+    const onChange = jest.fn()
 
-  await wrapper.setProps({ disabled: false, readonly: true })
-  await wrapper.find('.var-rate__content').trigger('click')
-  expect(wrapper.html()).toMatchSnapshot()
-  expect(onChange).toHaveBeenCalledTimes(0)
-})
+    const wrapper = mount(VarRate, {
+      props: {
+        disabled: true,
+        readonly: false,
+        onChange,
+      },
+    })
 
-test('test rate count', () => {
-  const wrapper = mount(VarRate, {
-    props: {
-      count: 10,
-    },
+    await wrapper.find('.var-rate__content').trigger('click')
+    expect(onChange).toHaveBeenCalledTimes(0)
+
+    await wrapper.setProps({ disabled: false, readonly: true })
+    await wrapper.find('.var-rate__content').trigger('click')
+    expect(onChange).toHaveBeenCalledTimes(0)
+
+    wrapper.unmount()
   })
 
-  expect(wrapper.findAll('.var-rate__content').length).toBe(10)
-  expect(wrapper.html()).toMatchSnapshot()
+  test('test rate count', () => {
+    const wrapper = mount(VarRate, {
+      props: {
+        count: 10,
+      },
+    })
 
-  wrapper.unmount()
-})
+    expect(wrapper.findAll('.var-rate__content').length).toBe(10)
 
-test('test rate gap', () => {
-  const wrapper = mount(VarRate, {
-    props: {
-      gap: 10,
-    },
+    wrapper.unmount()
   })
 
-  expect(wrapper.html()).toMatchSnapshot()
+  test('test rate gap', () => {
+    const wrapper = mount(VarRate, {
+      props: {
+        gap: 10,
+      },
+    })
 
-  wrapper.unmount()
+    console.log('wrapper.element', wrapper.element)
```

Comment:
It seems this test case has not been completed.

Reply:
I don't know how to compare its `gap` correctly.

Reply:
The `gap` prop actually adds `margin-right` for `var-rate__content`, so we can test it like this:

```
const el = wrapper.find('.var-rate__content')

expect(el.attributes('style')).toMatch('margin-right: 10px')
```

Reply:
it's work.But i'm not sure the Reg is Conforming to the specification